src/components/Tokens/TokenDetails/ChartSection.tsx

```diff
@@ -51,55 +52,52 @@ const TokenActions = styled.div`
   gap: 16px;
   color: ${({ theme }) => theme.textSecondary};
 `
-const NetworkBadge = styled.div<{ networkColor?: string; backgroundColor?: string }>`
-  border-radius: 5px;
-  padding: 4px 8px;
-  font-weight: 600;
-  font-size: 12px;
-  line-height: 12px;
-  color: ${({ theme, networkColor }) => networkColor ?? theme.textPrimary};
-  background-color: ${({ theme, backgroundColor }) => backgroundColor ?? theme.backgroundSurface};
-`
 
-export default function ChartSection({ token, tokenData }: { token: Token; tokenData: SingleTokenData | undefined }) {
-  const { chainId: connectedChainId } = useWeb3React()
+export function useTokenLogoURI(
+  token: NonNullable<SingleTokenData> | NonNullable<TopToken>,
+  nativeCurrency?: Token | NativeCurrency
+) {
+  const checksummedAddress = isAddress(token.address)
+  const chainId = CHAIN_NAME_TO_CHAIN_ID[token.chain]
+  return (
+    useCurrencyLogoURIs(nativeCurrency)[0] ??
+    (checksummedAddress && getTokenLogoURI(checksummedAddress, chainId)) ??
+    token.project?.logoUrl
+  )
+}
+
+export default function ChartSection({
+  token,
+  nativeCurrency,
+}: {
+  token: NonNullable<SingleTokenData>
+  nativeCurrency?: Token | NativeCurrency
+}) {
   const isFavorited = useIsFavorited(token.address)
   const toggleFavorite = useToggleFavorite(token.address)
-  const chainInfo = getChainInfo(token?.chainId)
-  const networkLabel = chainInfo?.label
-  const networkBadgebackgroundColor = chainInfo?.backgroundColor
-  const warning = checkWarning(token.address)
-
-  let currency = useCurrency(token.address)
+  const chainId = CHAIN_NAME_TO_CHAIN_ID[token.chain]
+  const L2Icon = getChainInfo(chainId).circleLogoUrl
+  const warning = checkWarning(token.address ?? '')
 
-  if (connectedChainId) {
-    const wrappedNativeCurrency = WRAPPED_NATIVE_CURRENCY[connectedChainId]
-    const isWrappedNativeToken = wrappedNativeCurrency?.address === token?.address
-    if (isWrappedNativeToken) {
-      currency = nativeOnChain(connectedChainId)
-    }
-  }
+  const { prices } = useTokenPricesCached(token)
 
-  const tokenName = tokenData?.name ?? token?.name
-  const tokenSymbol = tokenData?.tokens?.[0]?.symbol ?? token?.symbol
+  const logoSrc = useTokenLogoURI(token, nativeCurrency)
 
   return (
     <ChartHeader>
       <TokenInfoContainer>
         <TokenNameCell>
-          <CurrencyLogo currency={currency} size={'32px'} symbol={tokenSymbol} />
-          {tokenName ?? <Trans>Name not found</Trans>}
-          <TokenSymbol>{tokenSymbol ?? <Trans>Symbol not found</Trans>}</TokenSymbol>
+          <LogoContainer>
+            <CurrencyLogo src={logoSrc} size={'32px'} symbol={nativeCurrency?.symbol ?? token.symbol} />
+            <L2NetworkLogo networkUrl={L2Icon} size={'16px'} />
+          </LogoContainer>
+          {nativeCurrency?.name ?? token.name ?? <Trans>Name not found</Trans>}
```

Comment:
what does this failure case look like? if you're landing on a page with an address but we can't fetch the token name or symbol, doesn't that mean the token doesn't exist? so how could you get there?

Reply:
I think this is definitely something to make another PR/ticket for if there isn't already, figure out which data points we require else return. hopefully relay's @required also helps here tho, as it should throw out results that don't have required fields